src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -34,29 +34,60 @@ public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstruct
         /// </summary>
         private BoundExpression RewriteDeconstruction(BoundTupleExpression left, Conversion conversion, BoundExpression right, bool isUsed)
         {
+            var lhsTemps = ArrayBuilder<LocalSymbol>.GetInstance();
+            var lhsEffects = ArrayBuilder<BoundExpression>.GetInstance();
+            ArrayBuilder<Binder.DeconstructionVariable> lhsTargets = GetAssignmentTargetsAndSideEffects(left, lhsTemps, lhsEffects);
+            BoundExpression result = RewriteDeconstruction(lhsTargets, conversion, left.Type, right, isUsed);
+            Binder.DeconstructionVariable.FreeDeconstructionVariables(lhsTargets);
+            if (result is null)
+            {
+                lhsTemps.Free();
+                lhsEffects.Free();
+                return null;
+            }
+
+            return _factory.Sequence(lhsTemps.ToImmutableAndFree(), lhsEffects.ToImmutableAndFree(), result);
+        }
+
+        private BoundExpression RewriteDeconstruction(
+            ArrayBuilder<Binder.DeconstructionVariable> lhsTargets,
+            Conversion conversion,
+            TypeSymbol leftType,
+            BoundExpression right,
+            bool isUsed)
+        {
+            if (right.Kind == BoundKind.ConditionalOperator)
+            {
+                var conditional = (BoundConditionalOperator)right;
+                return conditional.Update(
+                    conditional.IsRef,
```

Comment:
>conditional.IsRef [](start = 20, length = 17)

I think you had an assertion for `!conditional.IsRef`. I thought that was a good idea. Consider restoring.